plugins/module_utils/foreman_helper.py

```diff
@@ -388,21 +388,23 @@ def find_resource(self, resource, search, name=None, params=None, failsafe=False
                 result = self.show_resource(resource, result['id'], params=params)
         return result
 
-    def find_resource_by_name(self, resource, name, **kwargs):
-        search = 'name="{0}"'.format(name)
-        kwargs['name'] = name
+    def find_resource_by(self, resource, search_field, value, **kwargs):
```

Comment:
I'm not sure I think this indirection is a great added value. Foreman is very much based on the search strings. The API is not obvious. Previously it was also possible to search using multiple fields at the same time which this API doesn't allow.

Reply:
But `find_resource` still exists, this only combines the `find_resource_by_*` helpers, which never accepted more than one field.

Reply:
I was thinking about a `build_search_string` helper so you don't need so many versions. In the end I wasn't sure if it would really simplify things.